src/opensynth/models/faraday/new_gmm/new_gmm_model.py

```diff
@@ -281,59 +284,79 @@ def training_step(self, batch) -> None:
         # Run e-step
         log_prob, log_resp = self.gmm_module.e_step(encoded_batch)
         # Run m-step
-        precision_cholesky, weights, means, covar = self.gmm_module.m_step(
-            encoded_batch, log_resp
+        precision_cholesky, weights, means, covariances = (
+            self.gmm_module.m_step(encoded_batch, log_resp)
         )
         # Update model params. This only updates the params
         # on the current device
         self.gmm_module.update_params(
             weights=weights,
             means=means,
             precision_cholesky=precision_cholesky,
-            covariances=covar,
+            covariances=covariances,
             log_prob=log_prob,
         )
 
+        if self.compute_on_batch:
+            # forward performs update, compute and reset metrics
```

Comment:
I think you should use this torch lightning callback:
- https://lightning.ai/docs/pytorch/stable/extensions/callbacks.html#on-train-batch-end

```python
def on_train_batch_end(self, args...) -> None:
    if self.compute_on_batch:
          metrics.forward()
          self.gmm_module.update_params()